libs/ledger-live-common/src/hw/getBatteryStatus.ts

```diff
@@ -22,38 +22,55 @@ enum FlagMasks {
   ISSUE_TEMPERATURE = 0x00000020,
 }
 
-const getBatteryStatus = async (
+// conditional type used to infer what will be the tuple of returned values of the getBatteryStatus
+// command. According to the tuple that of StatusTypes that is requested, it will infer the corresponding
+// tupple of either number or BatteryStatusFlags for the result
+type BatteryStatusTuple<Statuses extends ReadonlyArray<BatteryStatusTypes>> = {
+  [index in keyof Statuses]: Statuses[index] extends BatteryStatusTypes.BATTERY_FLAGS
+    ? BatteryStatusFlags
+    : number;
+};
+
+const getBatteryStatus = async <
+  StatusesType extends ReadonlyArray<BatteryStatusTypes>
+>(
```

Comment:
I think the associated unit tests are failing

```
pnpm common jest libs/ledger-live-common/src/hw/getBatteryStatus.test.ts
```

Reply:
Yeah, I had forgotten to adapt the unit tests to the changes I've made. Good catch. Done